x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/EsRelation.java

```diff
@@ -35,30 +38,41 @@ public class EsRelation extends LeafPlan {
         EsRelation::readFrom
     );
 
-    private final EsIndex index;
-    private final List<Attribute> attrs;
-    private final boolean frozen;
+    private final String indexName;
     private final IndexMode indexMode;
+    private final Map<String, IndexMode> indexNameWithModes;
+    private final List<Attribute> attrs;
 
-    public EsRelation(Source source, EsIndex index, IndexMode indexMode, boolean frozen) {
-        this(source, index, flatten(source, index.mapping()), indexMode, frozen);
-    }
-
-    public EsRelation(Source source, EsIndex index, List<Attribute> attributes, IndexMode indexMode) {
-        this(source, index, attributes, indexMode, false);
+    public EsRelation(Source source, EsIndex index, IndexMode indexMode) {
+        this(source, index.name(), indexMode, index.indexNameWithModes(), flatten(source, index.mapping()));
     }
 
-    public EsRelation(Source source, EsIndex index, List<Attribute> attributes, IndexMode indexMode, boolean frozen) {
+    public EsRelation(
+        Source source,
+        String indexName,
+        IndexMode indexMode,
```

Comment:
I believe the `indexMode` attribute has become obsolete. It's required in the `UnresolvedRelation`, but if we keep track of `indexNameWithModes`, the latter has more complete information _obtained from actual field caps calls_ and should be preferred.

I think we only use this `indexMode` to check if a relation corresponds to a lookup index. I think we could just have a helper function instead `isLookupIndex` which checks the `indexNameWithModes`.

Reply:
Let me try to replace its usages with the map

Reply:
We also have 2 checks for `TIME_SERIES`:

https://github.com/elastic/elasticsearch/blob/ab2607783a18daf375f61c24b93fc2294c653608/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/physical/local/ReplaceSourceAttributes.java#L35

https://github.com/elastic/elasticsearch/blob/d953079380243f518ca3847fdcb4e1816a4c8163/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/EsPhysicalOperationProviders.java#L184

that is originally configured in:

https://github.com/elastic/elasticsearch/blob/212e16bc3a5f6bef412f2ba158d5ace7f62ea300/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java#L484

I am thinking of deferring that to a separate change

Reply:
I opened an issue for that https://github.com/elastic/elasticsearch/issues/120210